script/github-actions/validate-build-status.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable no-console */
+/* eslint-disable camelcase */
+const fetch = require('node-fetch');
+
+const args = process.argv.slice(2);
+const repo = args[0];
+const timeout = 2; // minutes
+const ymlName = args[1]; // TODO: Once e2e is changed to continous integration, we can remove
+const releaseSHA = args[2];
+let page = 1;
+let checkWorkflowURL = `https://api.github.com/repos/${repo}/actions/workflows/${ymlName}.yml/runs?branch=master&page=${page}&per_page=50`;
+
+/**
+ * fetch request for github action URL provided
+ * @param {string} url
+ */
+function getLatestWorkflow(url) {
+  const headers = { Accept: 'application/vnd.github.v3+json' };
+  return fetch(url, headers)
+    .then(response => {
+      if (!response.ok) {
+        return Promise.reject(
+          Error(`Response not okay with ${url}. Aborting.`),
+        );
+      }
+      return response.json();
+    })
+    .then(({ workflow_runs }) => {
+      if (workflow_runs.length === 0) {
+        return Promise.reject(Error(`No workflows returns. Aborting.`));
+      }
+
+      let validWorkflow;
+
+      // If SHA passed, get workflow information. Otherwise get the most recent
+      if (releaseSHA) {
+        validWorkflow = workflow_runs.find(
+          ({ head_sha }) => head_sha === releaseSHA,
+        );
+        if (validWorkflow === undefined) {
```

Comment:
A general falsey check should suffice, I think.

```suggestion
        if (!validWorkflow) {
```